### DISCUSSION TITLE
我想要使服务端出错时，展示前端404页面

### DISCUSSION THREAD

**Discussion by wmasfoe** [nuxt/nuxt#10189] [Questions (Nuxt 2)]
在服务端定义了 `/rest/test` 路由，当访问 `/rest/other` 时，展示前端的404页面。
前端配置了 `*.vue` 页面，用来做兜底。

前端这边是通过 `asyncData` 中的 `error({ statusCode: 404 })` 实现的；现在需要在 node 层实现这个功能

需求：
- 路由不进行跳转（不跳转到 404.vue ）
- 页面响应状态码为 404

尝试过的方法（服务端 node 框架为：express）：
- `res.redirect('/404')`，页面状态码为 404 页面展示的也是前端定义的 *.vue 页面，但是会改变浏览器地址栏，不符合需求第一条。
- `res.status(404)`，network 一直为 loading 状态
- 
  ```js
  res.status(404)
  res.send('')
  ```
> 这样就会展示浏览器自带的错误页
![image](https://user-images.githubusercontent.com/61968242/148364484-a841f854-2843-4d55-956b-b18788b61619.png)

我要怎样做才可以实现需求中的功能。

任何一条建议都对我有帮助，谢谢

**Comment by wmasfoe** [comment]
使用正则表达式实现了，谢谢

**Reply by yanqc1996** [reply]
> 使用正则表达式实现了，谢谢

想请教下这个是怎么实现的，遇到了同样的问题

**Reply by SKhler** [reply]
> > 使用正则表达式实现了，谢谢
> 
> 
> 
> 想请教下这个是怎么实现的，遇到了同样的问题
